### ISSUE TITLE
Remove font-variant @font-face descriptor from Fonts 3

### ISSUE THREAD

**Issue by litherum** [w3c/csswg-drafts#2531]
WebKit is the only implementation, and our implementation has serious architectural-level issues. Given we are the only implementation, we should remove this descriptor from the spec (and remove the busted support from WebKit)

**Comment by faceless2** [comment]
Although our implementation isn't ready for public release yet, we **do** have full support for this, and we have recently added some test-cases to reflect this at https://github.com/w3c/web-platform-tests/pull/9397, with accompanying discussion at https://github.com/w3c/csswg-drafts/issues/2265.

**Comment by astearns** [comment]
@faceless2 in the minutes above, “until we talk” means until we talk to you. We aren’t yet removing this, just postponing deciding whether it’s specified to the next module level.

**Comment by faceless2** [comment]
Thank you for your consideration! But no real need. All I wanted to add was that there was another implementation out there, as I'm aware that that can make a difference to the decision process. Yes, ours is (or will be) a PDF renderer, along the lines of Prince or PDF Reactor.

We do have a working implementation of this part of the spec as it stands, although (IMHO) the spec could be modified to better serve the purpose it was intended for - some of this was covered in the issue I referenced and I know everyone is aware of those.

To sum up: it doesn't bother us in the slightest deferring this until L4, please do.

**Comment by svgeesus** [comment]
I made an equivalence test for font-variant descriptor vs. property
https://www.w3.org/People/chris/fwf/font-variant-descriptor-01.html
https://www.w3.org/People/chris/fwf/font-variant-descriptor-01-ref.html

Now merged into WPT
https://github.com/w3c/web-platform-tests/pull/10415
and in CSS WG test harness
https://test.csswg.org/harness/test/css-fonts-3_dev/single/font-variant-descriptor-01/

Chrome (Canary) Firefox (Nightly) fail as they don't implement the descriptor. Safari (Tech Preview) passes the descriptor subtests but fails one of the property subtests. Edge (17 preview) fails.

**Comment by svgeesus** [comment]
@litherum could you please review the [trivial test change](https://github.com/w3c/web-platform-tests/pull/11035) which just re-assigns that test to Fonts 4. I checked that all the links correctly resolve in Fonts 4.

This is one of only [two non-passing tests for Fonts 3](https://test.csswg.org/harness/results/css-fonts-3_dev/grouped/filter/1/) at this point.

**Comment by svgeesus** [comment]
This test got moved to the [Fonts 4 testsuite](https://test.csswg.org/harness/test/css-fonts-4_dev/single/font-variant-descriptor-01/). There are still [zero passing implementations](https://test.csswg.org/harness/details/css-fonts-4_dev/font-variant-descriptor-01/)

**Comment by svgeesus** [comment]
@litherum besides the "remove from Fonts 3" component (already done) I also get the impression this issue is "should it be there at all" so I am leaving it open. Can you confirm?

**Comment by svgeesus** [comment]
Still [no passing implementations](https://test.csswg.org/harness/details/css-fonts-4_dev/font-variant-descriptor-01/) for font-variant descriptor. Do we want to keep this? Is there implementer interest?

**Comment by drott** [comment]
At this point we don't implement other descriptors in @font-face, such as `font-feature-settings` and `font-variation-settings` - @litherum, do you suggest to keep those and only remove `font-variant`?

**Comment by litherum** [comment]
Let's remove them all, if possible!!!

We may have compat problems, because we've been shipping the other ones for a while. I'm willing to try it, though!

**Comment by svgeesus** [comment]
@drott wrote

> At this point we don't implement other descriptors in @font-face, such as font-feature-settings and font-variation-settings - @litherum, do you suggest to keep those and only remove font-variant?

I prefer to discuss those in a separate issue.

**Comment by drott** [comment]
> I prefer to discuss those in a separate issue.

Makes sense, let's track those separately.

**Comment by heycam** [comment]
This means we should also remove `variant` from the `FontFace` interface in the Font Loading API spec.

**Comment by litherum** [comment]
@heycam only edited the font loading spec, not the fonts spec.

**Comment by litherum** [comment]
Oh, I see, @svgeesus did it in d3db8697852a6ccc1c8dafcb4aff1f516198bb50 in fonts-4, and it was never in fonts-3 (meaning this issue is incorrectly titled).

**Comment by litherum** [comment]
Wow, it looks like I accidentally removed the relevant section from the "Feature and Variation Precedence" section in https://github.com/w3c/csswg-drafts/commit/0170cbc76ae2d772758a964856de7f5963f9114d. Oops! Good thing we eventually deleted it anyway!
